test/models/device.test.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const device = require('../../models/Device');
+
+
+describe('When given a name and id', () => {
+  it('returns object with name, id, and _id', () => {
+    const result = device({ name: 'test-name', id: 'test-id' });
+    expect(result.name).toBe('test-name');
+    expect(result.id).toBe('test-id');
+    expect(result._id).toEqual(expect.stringMatching(/[0-9a-f]{24}/g));
+  });
+});
+
+describe('When given a name and no id', () => {
+  it('returns \'No ID provided.\'', () => {
+    const result = device({ name: 'test-name' });
+    expect(result).toBe('No ID provided.');
```

Comment:
I think we discussed consistent return after you made this PR: return an object with a lone error key when there are errors, instead of a string. That way in the calling function you can use `if (result.error) {...}` to detect errors and process their innards.